### DISCUSSION TITLE
COLMAP not using NVIDIA A100 GPU

### DISCUSSION THREAD

**Discussion by vishal-lokare** [colmap/colmap#2256] [Q&A]
Hey everyone, 

I am using a VM in a private cloud with following specs

Ubuntu 20.04
![image](https://github.com/colmap/colmap/assets/31512838/602a4754-a77f-4a51-8310-93d6cf242d5c)

CPU - AMD EPYC 7413 24-core 
RAM - around 200GB
GPU - NVIDIA A100 40GB PCIe

Latest COLMAP is installed using the instructions provided in [https://colmap.github.io/install.html#linux](https://colmap.github.io/install.html#linux)

CUDA support is available as tested from torch - 
![image](https://github.com/colmap/colmap/assets/31512838/660ed5e9-30cd-4086-bb4e-9e6b674502f8)

But still, the whole process of making a sparse and a dense model takes around 1 hour for 700 images.

While a reconstruction is happening - 
![image](https://github.com/colmap/colmap/assets/31512838/d4406e99-9d1b-42e7-9ae3-7f0fd7e3667a)

![image](https://github.com/colmap/colmap/assets/31512838/cbc60485-119e-4f07-963b-91554ba8739a)

What am I missing here? I am very new to this so any help will be appreciated.

**Comment by ahojnnes** [comment]
Did you compile Colmap with cuda support? You should be able to confirm that when typing "colmap --help" on the command line at the top of the output.

**Reply by vishal-lokare** [reply]
![image](https://github.com/colmap/colmap/assets/31512838/f264471b-fd09-4297-9522-7a623fa3b7e4)

Yes Johannes, it says "without CUDA". Can you guide me on how to install it with CUDA support.

**Reply by ahojnnes** [reply]
Make sure to install cuda on your system and then call cmake as "cmake -DCUDA_ENABLED=ON ...". You should see that cuda was found and also the specific version. Basic instructions are also in the documentation.
